### PR TITLE
Improve k-means algorithm

### PR DESCRIPTION
**Pull request by gregori0o**
Add interface for options classes.
Add new parameters to k-means algorithm.
Implementation of k-means++.

Edit: Add Label for description.

### CODE REVIEW

File: src/algorithms/clustering/k_means.py [hunk 1]
```diff
@@ -1,23 +1,25 @@
 import pandas as pd
 import numpy as np
-from typing import List, Tuple, Generator, Union
+from typing import List, Tuple, Union, Optional
 
-num_types = [int, float, np.float, np.int]
+initial_types = ['random', 'kmeans++']
```
Comment (mhawryluk) [inline src/algorithms/clustering/k_means.py]: makes it sound as though the types are initial. maybe switch to initialization_types, initialize_types or init_types

File: src/widgets/options_widgets/k_means_options.py [hunk 2]
```diff
@@ -1,27 +1,46 @@
...
+        self.num_steps_spinbox.setMaximum(1000)
+        self.num_steps_spinbox.setSpecialValueText('no limit')
+        self.num_steps_spinbox.setValue(0)
+        self.layout.addRow(QLabel("Maximum number of iterations:"), self.num_steps_spinbox)
+
+        self.num_repeat_spinbox = QSpinBox()
+        self.num_repeat_spinbox.setMinimum(1)
+        self.num_repeat_spinbox.setMaximum(100)
+        self.num_repeat_spinbox.setValue(1)
+        self.layout.addRow(QLabel("Number of repeating:"), self.num_repeat_spinbox)
```
Comment (mhawryluk) [inline src/widgets/options_widgets/k_means_options.py]:
```suggestion
        self.layout.addRow(QLabel("Number of repetitions:"), self.num_repeat_spinbox)
```

**Review by mhawryluk — APPROVED** [review]
Great! 💃

File: src/widgets/options_widgets/k_means_options.py [hunk 3]
```diff
@@ -1,27 +1,46 @@
...
+
     def get_data(self) -> dict:
         return {
             'num_clusters': self.num_clusters_spinbox.value(),
-            'metrics': self.metrics_spinbox.value()
+            'metrics': self.metrics_spinbox.value(),
+            'repeats': self.num_repeat_spinbox.value(),
+            'iterations': None if self.num_steps_spinbox.value() == 0 else self.num_steps_spinbox.value(),
```
Comment (t0maboro) [inline src/widgets/options_widgets/k_means_options.py]:
```suggestion
            'iterations': self.num_steps_spinbox.value() or None,
```
Not important if you think that it's less readable or something else. Your one is nice too!

**Review by t0maboro — APPROVED** [review]
Nice job! Left one comment, but it should not prevent you from merging code in actual form if you wish.

**Pull request merged**